iceoryx_utils/include/iceoryx_utils/internal/concurrent/lockfree_queue/cyclic_index.inl

```diff
@@ -91,7 +91,7 @@ bool CyclicIndex<CycleLength, ValueType>::isOneCycleBehind(const CyclicIndex& ot
 template <uint64_t CycleLength, typename ValueType>
 int64_t CyclicIndex<CycleLength, ValueType>::operator-(const CyclicIndex<CycleLength, ValueType>& rhs) const
 {
-    return m_value - rhs.m_value;
+    return static_cast<int64_t>(m_value) - static_cast<int64_t>(rhs.m_value);
```

Comment:
I think it's better to first subtract and then cast. The underflow is well defined for `uint64_t` but not for `int64_t`. @MatthiasKillat your opinion?